source/speechDictHandler/__init__.py

```diff
@@ -115,9 +114,27 @@ def sub(self, text):
 				del self[index]
 		return text
 
+
+# useful for #14689 (see below)
+_IGNORE_TRAILING_WHITESPACE_LENGTH = 100
+
 def processText(text):
 	if not globalVars.speechDictionaryProcessing:
 		return text
+	# #14689: API level < 2 UIA consoles have many blank lines, which slows processing to a halt
+	focus = globalVars.focusObject
+	try:
+		# get TextInfo implementation for object review mode
+		textInfo, obj = review.getObjectPosition(focus)
+	except AttributeError:  # no makeTextInfo
+		textInfo = None
+	# late import to prevent circular dependency
+	# ConsoleUIATextInfo is shared by apiLevel 1 and 0 implementations (not 2)
+	from NVDAObjects.UIA.winConsoleUIA import ConsoleUIATextInfo
+	if isinstance(textInfo, ConsoleUIATextInfo):
+		stripText = text.rstrip()
+		if len(text) - len(stripText) > _IGNORE_TRAILING_WHITESPACE_LENGTH:

```

Comment:
I really think it is easier to keep these together, mainly for readibility.

```suggestion
		IGNORE_TRAILING_WHITESPACE_LENGTH = 100
		if len(text) - len(stripText) > IGNORE_TRAILING_WHITESPACE_LENGTH:
```